src/engraving/libmscore/beam.h

```diff
@@ -33,20 +33,107 @@ class ChordRest;
 class Factory;
 class Skyline;
 class System;
-
+class Beam;
 enum class ActionIconType;
 enum class SpannerSegmentType;
 
-struct BeamFragment;
+//---------------------------------------------------------
+//   BeamFragment
+//    position of primary beam
+//    idx 0 - DirectionV::AUTO or DirectionV::DOWN
+//        1 - DirectionV::UP
+//---------------------------------------------------------
+
+struct BeamFragment {
+    double py1[2];
+    double py2[2];
+};
 
 struct BeamSegment {
     mu::LineF line;
-    int level;
-    bool above; // above level 0 or below? (meaningless for level 0)
+    int level = 0;
+    bool above = false; // above level 0 or below? (meaningless for level 0)
     Fraction startTick;
     Fraction endTick;
 };
 
+struct TremAnchor {
+    ChordRest* chord1 = nullptr;
+    double y1 = 0.;
+    double y2 = 0.;
+};
+
+class BeamLayout
+{
+private:
+    static constexpr std::array _maxSlopes = { 0, 1, 2, 3, 4, 5, 6, 7 };
```

Comment:
As far I can tell, this array isn't used in this file, so it's better to declare it directly in cpp

Reply:
done!